### PR TITLE
rbd: update rbd sc for topology provisioning

### PR DESCRIPTION
**Pull request by parth-gr**
with new fix at csi ceph/ceph-csi#4459 we dont need the pool parameter dependency from the storageclass So removing it

### CODE REVIEW

**Review by Madhu-1 — COMMENTED** [review]
LGTM, please test the changes to ensure it works as expected

**Comment by parth-gr** [comment]
Rook Pr should get merged first https://github.com/rook/rook/pull/13977, as it has the json changes

**Review by malayparida2000 — APPROVED** [review]
/lgtm
Please ensure testing this before merge.

cc @iamniting

**Review by iamniting — APPROVED** [review]
/hold

@parth-gr adding a hold for testing pls remove it once you are done with the testing.

Also pls look at the external cluster case as well.

**Comment by parth-gr** [comment]
There is a bug for internal mode, adding a fix in second commit
Tested:
```
parameters:
  clusterID: openshift-storage
  csi.storage.k8s.io/controller-expand-secret-name: rook-csi-rbd-provisioner
  csi.storage.k8s.io/controller-expand-secret-namespace: openshift-storage
  csi.storage.k8s.io/fstype: ext4
  csi.storage.k8s.io/node-stage-secret-name: rook-csi-rbd-node
  csi.storage.k8s.io/node-stage-secret-namespace: openshift-storage
  csi.storage.k8s.io/provisioner-secret-name: rook-csi-rbd-provisioner
  csi.storage.k8s.io/provisioner-secret-namespace: openshift-storage
  imageFeatures: layering,deep-flatten,exclusive-lock,object-map,fast-diff
  imageFormat: "2"
  topologyConstrainedPools: |-
    [
      {
        "poolName": "ocs-storagecluster-cephblockpool-rack0",
        "domainSegments": [
          {
            "domainLabel": "rack",
            "value": "rack0"
          }
        ]
      },
      {
        "poolName": "ocs-storagecluster-cephblockpool-rack1",
        "domainSegments": [
          {
            "domainLabel": "rack",
            "value": "rack1"
          }
        ]
      },
      {
        "poolName": "ocs-storagecluster-cephblockpool-rack2",
        "domainSegments": [
          {
            "domainLabel": "rack",
            "value": "rack2"
          }
        ]
      }
    ]

```

**Comment by malayparida2000** [comment]
> There is a bug for internal mode, adding a fix in second commit Tested:
> 
> ```
> parameters:
>   clusterID: openshift-storage
>   csi.storage.k8s.io/controller-expand-secret-name: rook-csi-rbd-provisioner
>   csi.storage.k8s.io/controller-expand-secret-namespace: openshift-storage
>   csi.storage.k8s.io/fstype: ext4
>   csi.storage.k8s.io/node-stage-secret-name: rook-csi-rbd-node
>   csi.storage.k8s.io/node-stage-secret-namespace: openshift-storage
>   csi.storage.k8s.io/provisioner-secret-name: rook-csi-rbd-provisioner
>   csi.storage.k8s.io/provisioner-secret-namespace: openshift-storage
>   imageFeatures: layering,deep-flatten,exclusive-lock,object-map,fast-diff
>   imageFormat: "2"
>   topologyConstrainedPools: |-
>     [
>       {
>         "poolName": "ocs-storagecluster-cephblockpool-rack0",
>         "domainSegments": [
>           {
>             "domainLabel": "rack",
>             "value": "rack0"
>           }
>         ]
>       },
>       {
>         "poolName": "ocs-storagecluster-cephblockpool-rack1",
>         "domainSegments": [
>           {
>             "domainLabel": "rack",
>             "value": "rack1"
>           }
>         ]
>       },
>       {
>         "poolName": "ocs-storagecluster-cephblockpool-rack2",
>         "domainSegments": [
>           {
>             "domainLabel": "rack",
>             "value": "rack2"
>           }
>         ]
>       }
>     ]
> ```

Let's squash to a single commit, the bug is basically related to this only

**Comment by parth-gr** [comment]
Done

**Review by malayparida2000 — APPROVED** [review]
/lgtm

**Pull request merged**
